bin/disco_deploy.py

```diff
@@ -92,13 +93,20 @@ def run():
         test_aws = aws
 
     vpc = DiscoVPC.fetch_environment(environment_name=env)
+    bake = DiscoBake(config, aws.connection)
 
     deploy = DiscoDeploy(
-        aws, test_aws, DiscoBake(config, aws.connection), DiscoGroup(env), DiscoELB(vpc),
+        aws, test_aws, bake, DiscoGroup(env), DiscoELB(vpc),
         pipeline_definition=pipeline_definition,
         ami=args.get("--ami"), hostclass=args.get("--hostclass"),
         allow_any_hostclass=args["--allow-any-hostclass"])
 
+    if args["--ami"] and args["--hostclass"]:
```

Comment:
You could move the test comparing the ami and the hostclass before initializing 'deploy'